src/driver-roccat-kone-aimo.c

```diff
@@ -0,0 +1,876 @@
+
+	settings_report->checksum = roccat_compute_crc(buf, ROCCAT_REPORT_SIZE_SETTINGS);
+
+	rc = ratbag_hidraw_set_feature_report(device, ROCCAT_REPORT_ID_SETTINGS,
+					      buf, ROCCAT_REPORT_SIZE_SETTINGS);
+
+	if (rc < 0)
+		return rc;
+
+	if (rc != ROCCAT_REPORT_SIZE_SETTINGS)
+		return -EIO;
+
+	rc = roccat_wait_ready(device);
+	if (rc)
+		log_error(device->ratbag,
+			  "Error while waiting for the device to be ready: %s (%d)\n",
+			  strerror(-rc), rc);
+
+	return rc;
+}
+
+static void
+roccat_read_profile(struct ratbag_profile *profile)
+{
+	struct ratbag_device *device = profile->device;
+	struct roccat_data *drv_data;
+	struct ratbag_resolution *resolution;
+	struct ratbag_button *button;
+	struct roccat_settings_report *setting_report;
+	uint8_t *buf;
+	unsigned int report_rate;
+	int dpi_x, dpi_y;
+	int rc;
+	unsigned int report_rates[] = { 125, 250, 500, 1000 };
+
+	assert(profile->index < ROCCAT_PROFILE_MAX);
+
+	drv_data = ratbag_get_drv_data(device);
+
+	setting_report = &drv_data->settings[profile->index];
+	buf = (uint8_t*)setting_report;
+	roccat_set_config_profile(device, profile->index, ROCCAT_CONFIG_SETTINGS);
+	rc = ratbag_hidraw_get_feature_report(device, ROCCAT_REPORT_ID_SETTINGS,
+					      buf, ROCCAT_REPORT_SIZE_SETTINGS);
+
+	if (rc < ROCCAT_REPORT_SIZE_SETTINGS)
+		return;
+
+	/* first retrieve the report rate, it is set per profile */
+	if (setting_report->report_rate < ARRAY_LENGTH(report_rates)) {
+		report_rate = report_rates[setting_report->report_rate];
+	} else {
+		log_error(device->ratbag,
+			  "error while reading the report rate of the mouse (0x%02x)\n",
+			  buf[26]);
+		report_rate = 0;
+	}
+
+	ratbag_profile_set_report_rate_list(profile, report_rates,
+					    ARRAY_LENGTH(report_rates));
+	ratbag_profile_set_report_rate(profile, report_rate);
+
+	ratbag_profile_for_each_resolution(profile, resolution) {
+		dpi_x = setting_report->xres[resolution->index] * 50;
+		dpi_y = setting_report->yres[resolution->index] * 50;
+		//if (!(setting_report->dpi_mask & (1 << resolution->index))) {
+		//	/* the profile is disabled, overwrite it */
+		//	dpi_x = 0;
+		//	dpi_y = 0;
+		//}
+
+		ratbag_resolution_set_resolution(resolution, dpi_x, dpi_y);
+		ratbag_resolution_set_cap(resolution,
+					  RATBAG_RESOLUTION_CAP_SEPARATE_XY_RESOLUTION);
+		resolution->is_active = (resolution->index == setting_report->current_dpi);
+
+		ratbag_resolution_set_dpi_list_from_range(resolution, ROCCAT_MIN_DPI, ROCCAT_MAX_DPI);
+	}
+
+	buf = drv_data->profiles[profile->index];
+	roccat_set_config_profile(device, profile->index, ROCCAT_CONFIG_KEY_MAPPING);
+	rc = ratbag_hidraw_get_feature_report(device, ROCCAT_REPORT_ID_KEY_MAPPING,
+					      buf, ROCCAT_REPORT_SIZE_PROFILE);
+
+	msleep(10);
+
+	if (rc < ROCCAT_REPORT_SIZE_PROFILE)
+		return;
+
+    ratbag_profile_for_each_button(profile, button)
+		roccat_read_button(button);
+
+    // No CRC check for Roccat Kone AIMO, but leaving for reference if needed for other models.
```

Comment:
But this driver is specific for the AIMO :stuck_out_tongue:. Please remove the unused CRC related code, we still have it on the other driver, so we can easily grab it if needed in the future.

Reply:
Removed